clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -420,7 +423,7 @@ boolean joinGroupIfNeeded(final Timer timer) {
                 // need to set the flag before calling onJoinPrepare since the user callback may throw
                 // exception, in which case upon retry we should not retry onJoinPrepare either.
                 needsJoinPrepare = false;
-                onJoinPrepare(generation.generationId, generation.memberId);
+                onJoinPrepare(generation.generationId, generation.memberId, pollTimer == null ? time.timer(0L) : pollTimer);
```

Comment:
I'm thinking the `pollTimer` should not be null here, which makes sense more. We should pass a non-null timer into methods.

Reply:
change codes already commit